File: server/src/main/java/com/ssafy/cafe/model/dto/Category.java
package com.ssafy.cafe.model.dto;

public enum Category {
	FICTION("FICTION"), 
	COOKING("COOKING"), 
	SCIENCE("SCIENCE+NATURE"),
	HEALTH("MEDICAL"),
	HIANDSOC("HISTORY+SOCIAL SCIENCE"),
	TRAVEL("TRAVEL"),
	CRANDHOANDFI("GARDENING+GAMES+PETS"),
	TEANDEN("TRANSPORTATION+ARCHITECTURE"),
	RELIGION("RELIGION+BIBLES"),
	POANDLI("POETRY+LITERARY COLLECTIONS"),
	ARTANDPER("ART+PERFORMING ARTS"),
	COMPUTER("COMPUTERS"),
	HUMAN("DESIGN+PHILOSOPHY+PSYCHOLOGY"),
	SELFHELF("Self-Help+DESIGN"),
	POANDSO("POLITICAL SCIENCE+SOCIAL SCIENCE+LAW"),
	FOREIGN("FOREIGN LANGUAGE STUDY"),
	JUVENILE("JUVENILE FICTION+");
	
	private final String name;
	
	private Category(String name) {
		this.name = name;
	}
	
	public String getName() {
		return name;
	}
	
    public static Category fromString(String text) {
        for (Category category : Category.values()) {
            if (category.name().equalsIgnoreCase(text)) {
                return category;
            }
        }
        throw new IllegalArgumentException("No constant with text " + text + " found");
    }
}

File: finalproject/app/src/main/java/com/ssafy/finalproject/Category.kt
package com.ssafy.finalproject

enum class Category(private val categoryKr: String) {
    FICTION("소설"),
    COOKING("요리"),
    SCIENCE("과학"),
    HEALTH("건강"),
    HIANDSOC("역사/문화"),
    TRAVEL("여행"),
    CRANDHOANDFI("취미/실용/스포츠"),
    TEANDEN("기술/공학"),
    RELIGION("종교"),
    POANDLI("시/에세이"),
    ARTANDPER("예술/대중문화"),
    COMPUTER("컴퓨터/IT"),
    HUMAN("인문"),
    SELFHELP("자기계발"),
    POANDSO("정치/사회"),
    FOREIGN("외국어"),
    JUVENILE("청소년");

    fun getCategoryKr(): String {
        return categoryKr
    }
}
